src/biogeophys/HillslopeHydrologyMod.F90

```diff
@@ -0,0 +1,1140 @@
+          if (use_hillslope_routing) then 
+                       
+             ! Total area occupied by each hillslope (m2) is
+             ! grc%area(g)*1.e6*lun%wtgcell(l)*pct_hillslope(l,nh)*0.01
+             ! Number of representative hillslopes per landunit
+             ! is the total area divided by individual area
+
+             lun%stream_channel_number(l) = 0._r8
+             do nh = 1, nhillslope
+                if(hillslope_area(nh) > 0._r8) then
+                   nhill_per_landunit(nh) = grc%area(g)*1.e6_r8*lun%wtgcell(l) &
+                        *pct_hillslope(l,nh)*0.01/hillslope_area(nh)
+
+                   lun%stream_channel_number(l) = lun%stream_channel_number(l) &
+                        + nhill_per_landunit(nh)
+                endif
+             enddo
+             
+             ! Calculate steam channel length
+             ! Total length of stream banks is individual widths
+             ! times number of hillslopes per landunit divided
+             ! by 2 to convert from bank length to channel length
+
+             lun%stream_channel_length(l) = 0._r8
+             do c = lun%coli(l), lun%colf(l)
+                if(col%cold(c) == ispval) then
+                   lun%stream_channel_length(l) = lun%stream_channel_length(l) &
+                        + col%hill_width(c) * nhill_per_landunit(col%hillslope_ndx(c))
+                endif
+             enddo
+             lun%stream_channel_length(l) = 0.5_r8 * lun%stream_channel_length(l)
+          endif
+                       
+          ! if missing hillslope information on surface dataset, fill data
+          ! and recalculate hillslope_area
+          if (sum(hillslope_area) == 0._r8) then
+             do c = lun%coli(l), lun%colf(l)
+                nh = col%hillslope_ndx(c)
+                col%hill_area(c) = (grc%area(g)/real(lun%ncolumns(l),r8))*1.e6_r8 ! km2 to m2
+                col%hill_width(c)    = sqrt(col%hill_area(c))
+                col%hill_slope(c)    = tan((rpi/180.)*col%topo_slope(c))
+                col%hill_aspect(c)   = (rpi/2.) ! east (arbitrarily chosen)
+                if (nh > 0) then
+                   col%hill_elev(c)     = col%topo_std(c) &
+                        *((c-lun%coli(l))/ncol_per_hillslope(nh))
+                   col%hill_distance(c) = sqrt(col%hill_area(c)) &
+                        *((c-lun%coli(l))/ncol_per_hillslope(nh))
+                   pct_hillslope(l,nh)  = 100/nhillslope
+                else
+                   col%hill_elev(c)     = col%topo_std(c)
+                   col%hill_distance(c) = sqrt(col%hill_area(c))
+                endif
+             enddo
+             
+          endif
+          
+          ! Recalculate column weights using input areas
+          check_weight = 0._r8
+          do c = lun%coli(l), lun%colf(l)
+             nh = col%hillslope_ndx(c)
+             if (col%is_hillslope_column(c)) then
+                col%wtlunit(c) = (col%hill_area(c)/hillslope_area(nh)) &
```

Comment:
(It may be that, in practice, this doesn't cause problems because the necessary routines are called in the course of doing the dynamic landunit code that I think is run – unnecessarily – every time step. But we don't want to rely on that behavior.)